pkg/virt-api/webhooks/validating-webhook/admitters/vms-admitter_test.go

```diff
@@ -1890,6 +1890,21 @@ var _ = Describe("Validating VM Admitter", func() {
 		Expect(resp.Result.Details.Causes[0].Type).To(Equal(metav1.CauseTypeFieldValueNotSupported))
 		Expect(resp.Result.Details.Causes[0].Message).To(Equal(fgMessage))
 	})
+
+	It("should reject VM with empty firmware UUID", func() {
+		vmi := libvmi.New()
+		vm := libvmi.NewVirtualMachine(vmi)
+		vm.Spec.Template.Spec.Domain.Firmware = &v1.Firmware{
```

Comment:
What if `vm.Spec.Template.Spec.Domain.Firmware` is `nil`? That would not fail in the current admitter IIUC.